src/main/java/com/hp/application/automation/tools/run/AbstractSvRunBuilder.java

```diff
@@ -0,0 +1,200 @@
+            return performImpl(build, launcher, listener);
+        } catch (Exception e) {
+            listener.fatalError(e.getMessage());
+            LOG.log(Level.SEVERE, "Build failed: " + e.getMessage(), e);
+        } finally {
+            double duration = (new Date().getTime() - startDate.getTime()) / 1000.;
+            logger.printf("Finished: %s in %.3f seconds%n%n", getDescriptor().getDisplayName(), duration);
+        }
+        return false;
+    }
+
+    protected void logConfig(PrintStream logger, String prefix) {
+        SvServiceSelectionModel ss = model.getServiceSelection();
+        switch (ss.getSelectionType()) {
+            case SERVICE:
+                logger.println(prefix + "Service name or id: " + ss.getService());
+                break;
+            case PROJECT:
+                logger.println(prefix + "Project path: " + ss.getProjectPath());
+                logger.println(prefix + "Project password: " + ((StringUtils.isNotBlank(ss.getProjectPassword())) ? "*****" : null));
+                break;
+            case ALL_DEPLOYED:
+                logger.println(prefix + "All deployed services");
+                break;
+            case DEPLOY:
+                logger.println(prefix + "Project path: " + ss.getProjectPath());
+                logger.println(prefix + "Project password: " + ((StringUtils.isNotBlank(ss.getProjectPassword())) ? "*****" : null));
+                logger.println(prefix + "Service name or id: " + ss.getService());
+                break;
+        }
+        logger.println(prefix + "Force: " + model.isForce());
+    }
+
+    protected List<ServiceInfo> getServiceList(boolean ignoreMissingServices, PrintStream logger, AbstractBuild<?, ?> build) throws Exception {
+        SvServiceSelectionModel s = getServiceSelection();
+        ICommandExecutor exec = createCommandExecutor();
+
+        ArrayList<ServiceInfo> res = new ArrayList<ServiceInfo>();
+
+        switch (s.getSelectionType()) {
+            case SERVICE:
+                addServiceIfDeployed(s.getService(), res, ignoreMissingServices, exec, logger);
+                break;
+            case PROJECT:
+                IProject project = loadProject(build);
+                for (IService svc : project.getServices()) {
+                    addServiceIfDeployed(svc.getId(), res, ignoreMissingServices, exec, logger);
+                }
+                break;
+            case ALL_DEPLOYED:
+                for (ServiceListAtom.ServiceEntry entry : exec.getServiceList(null).getEntries()) {
+                    res.add(new ServiceInfo(entry.getId(), entry.getTitle()));
+                }
+            case DEPLOY:
+                break;
+        }
+        return res;
+    }
+
+    private void addServiceIfDeployed(String service, ArrayList<ServiceInfo> results, boolean ignoreMissingServices, ICommandExecutor exec, PrintStream logger) throws CommunicatorException, CommandExecutorException {
```

Comment:
Ignore 1.
2. same issue as below - both extend same class and caught in general exception.